### DISCUSSION TITLE
404 when creating lte network via api or NMS

### DISCUSSION THREAD

**Discussion by HesselTjeerdsma** [magma/magma#5713] [Q&A] [answered]
Hello,

I am trying to deploy magmacore v1.4 to our on premise kubernetes cluster. So far everything is running, but I am not able to add a LTE network using the api directly or using the NMS. The LTE endpoints return the following error there:
```
{
  "message": "Not Found"
}
```
This is in the orc8r-nginx logs:
```
{"nginx.time_local": "24/Mar/2021:15:23:39 +0000","nginx.remote_addr": "10.42.0.1","nginx.request": "GET /magma/v1/lte HTTP/1.1","nginx.request_method": "GET","nginx.request_uri": "/magma/v1/lte","nginx.status": 404,"nginx.body_bytes_sent": 24,"nginx.request_length": 648,"nginx.request_time": 0.011,"nginx.server_name": "_","nginx.clientcert_port": 9070,"nginx.open_port": 9070,"nginx.client_serial": "08B8D713AAB8773CFD835D9AE5C51F0DBAB4B199","nginx.client_cn": "admin_operator"}
{"nginx.time_local": "24/Mar/2021:15:25:56 +0000","nginx.remote_addr": "10.42.0.1","nginx.request": "GET /magma/v1/lte HTTP/1.1","nginx.request_method": "GET","nginx.request_uri": "/magma/v1/lte","nginx.status": 404,"nginx.body_bytes_sent": 24,"nginx.request_length": 524,"nginx.request_time": 0.010,"nginx.server_name": "_","nginx.clientcert_port": 9070,"nginx.open_port": 9070,"nginx.client_serial": "<removed>","nginx.client_cn": "admin_operator"}
```
Next to that I can see this in the orc8r-eventd logs:
```
W0324 12:41:15.166901 1 service_config.go:197] Cannot find '/etc/magma/configs/magmad.yml': stat /etc/magma/configs/magmad.yml: no such file or directory, or Legacy Service Registry Configuration: '/etc/magma/magmad.yml': stat /etc/magma/magmad.yml: no such file or directory
```
Could somebody help with fixing this? Thanks!

**Comment by karthiksubraveti** [comment]
Hello @HesselTjeerdsma, from the kube pods you had posted. It doesn’t seem to be running lte module. You might have to deploy lte pods as well.(https://github.com/magma/magma/blob/31527f40d281b90ded08cdec2d2db661a6d31bb2/orc8r/cloud/deploy/terraform/orc8r-helm-aws/main.tf#L50. Following is the helm chart associated with LTE. https://github.com/magma/magma/blob/7e50344ff691e686bca77819bbcf361f067a5620/lte/cloud/helm/lte-orc8r/Chart.yaml) How did you do your deployment, was it through aws ?

**Comment by HesselTjeerdsma** [comment, marked as answer]
No, we are using a on-premise deployment consisting of multiple VMs on Hetzner Cloud. This is the helm repo we have been using `https://github.com/HesselTjeerdsma/magma-helm/blob/master/index.yaml` furthermore I used the [minikube ](https://docs.magmacore.org/docs/next/orc8r/dev_minikube_deployment)values file for deploying) which includes the magma-lte part, but not the orc8r-lte part. How would I deploy that? 
```
nms:
  secret:
    certs: orc8r-secrets-certs
  imagePullSecrets:
    - name: orc8r-secrets-registry
  pod:
    replicas:
      nginx:
        server: 1
      magmalte:
        server: 1
  magmalte:
    manifests:
      configmap: true
      secrets: true
      deployment: true
      service: true

```
We do not have the AGW running yet, is it possible that this is the issue?

I have now deployed the orc8r-lte helm chart, but it cannot find the orc8r_base image, which I have not uploaded during the build process. Where can I find this image or the Dockerfile for this image?

**Reply by karthiksubraveti** [reply]
AGW isn't required for creating LTE network. The image it references is actually the orc8r controller image. Can you point it to that image and check if this works ?

**Reply by HesselTjeerdsma** [reply]
Hi! Yeah, replacing `repository: <registry>/orc8r_base` to `repository: <registry>/controller` solved the issue. Thanks! I am able to add networks now. 

Could it be better to change `repository: <registry>/orc8r_base` to `repository: <registry>/controller`
in the [values.yaml](https://github.com/magma/magma/blob/master/lte/cloud/helm/lte-orc8r/values.yaml) as well? 

Let me know, then I will submit a merge request!

**Reply by karthiksubraveti** [reply]
Sounds good. Looks like we have referenced it in bunch of places https://github.com/magma/magma/search?q=orc8r_base. Can you fix all these places and add a comment to use the orc8r controller image from the registry. Thanks for fixing this.

**Reply by jdesai61ilu** [reply]
@HesselTjeerdsma  Can you please let us know the steps you followed to deploy orc8r-lte chart?

**Comment by HesselTjeerdsma** [comment]
This has been resolved already. Thanks for the follow up. You can close
this issue.
